### DISCUSSION TITLE
Set background 2d image ModelViewer Android

### DISCUSSION THREAD

**Discussion by Berna78** [google/filament#5603] [Q&A]
I would like to now how I can load image (png/jpg) from Resource/Asset and set it like "model" background, not using skybox. I try to set the modelviewer as transparent view but in this way it is placed in front of every other view. I would like to know if there is a way to set the background and make sure that the position on the z axis of the 3D model remains the default one?

**Comment by sdpopov-keyvariable** [comment]
1. It's possible to render several views between `\(begin|end)Frame`. So make the main view translucent, create a view with single quad textured with the background texture and orthographic camera, then render the background view first. 
2. Create quad node, set its priority to 0 and in material use `getWorldFromClipMatrix()` in vertex shader to make the quad always fill the screen.
3. Render frame with transparent background and place a system image view under the Filament view.

**Reply by Berna78** [reply]
The problem is that when rendering main view traslucent, this overlaps any graphic element present within the application (for example a navigation bar).
Also could you give me an example to refer to?

**Reply by sdpopov-keyvariable** [reply]
I guess the problem concerned with the OS view hierarchy (layout or order), not with Filament.

I haven't seen exact example. Unfortunately, it will take too much time to extract code. Anyway, the idea has already been explained.

**Reply by sdpopov-keyvariable** [reply]
PS. If you need for fill just part of the surface, then adjust the viewport property.

**Reply by romainguy** [reply]
> The problem is that when rendering main view traslucent, this overlaps any graphic element present within the application (for example a navigation bar).
> 
> Also could you give me an example to refer to?

It won't if you give the Filament view the proper dimensions and position. It doesn't have to be "full screen".

**Comment by romainguy** [comment]
For #2 you don't even need that. You can change the vertex domain of a material to be in clip space directly.

**Reply by Berna78** [reply]
Maybe I don't understand what you mean,  but in my application it is expected that there should be views overlaid on modelviewer.
Is there no other way to add a background image?
![Screenshot](https://user-images.githubusercontent.com/97038728/173875440-c8a0763d-3b41-408d-a65c-5af519833fbf.jpg)

**Reply by romainguy** [reply]
You can achieve this in two different ways:
- Don't use a SurfaceView, but a TextureView. That way you can have a regular background View behind the transparent TextureView and your other Views on top. It's less efficient than a SurfaceView though.
- Keep using a SurfaceView and draw your background inside the app either as a Skybox or as a custom Renderable: a fullscreen quad or triangle with a material that simply applies an unlit texture. Here's an example: https://github.com/google/filament/blob/main/samples/materials/image.mat The renderable (a fullscreen triangle) is created here: https://github.com/google/filament/blob/main/samples/image_viewer.cpp#L166

**Comment by yongwookim** [comment]
@romainguy @sdpopov-keyvariable Thank you very much for the guide. 
It would be extremely helpful for the community if you could elaborate a bit more about 1) how to fill custom created rectangle can always fill up the android screen 2) how a custom Renderable can stay back of the 3d scene all the time. I was able to put a quad with image texture but am having hard time to let it stay like wall paper even when camera rotates.

**Reply by nischaysingla** [reply]
Hi @yongwookim,
found any solution?

**Reply by yongwookim** [reply]
not quite. we've took workaround approach, 1) rendering filament on android textureview rather than surfaceview, 2) set it transparent background 3) have another textureview below filament.

**Reply by LeFrenchTouch** [reply]
Hi @yongwookim , any chance you could post a sample on how to make the TextureView transparent?
